### ISSUE TITLE
Scale to fit code removes embedded images

### ISSUE THREAD

**Issue by AndreKR** [BigBadaboom/androidsvg#236]
I'm trying to render [this SVG](https://www.yr.no/en/content/2-2192362/meteogram.svg) into an `ImageView` so that it fills the whole `ImageView` while keeping the aspect ratio.

The SVG has `width` and `height` but no `viewBox`, so  I added the recommended code (essentially the same case as "Dealing with Inkscape files"):

```kotlin
svg.setDocumentViewBox(0f, 0f, svg.documentWidth, svg.documentHeight);
svg.setDocumentHeight("100%")
svg.setDocumentWidth("100%")
```

Without the extra settings it renders like this:

![image](https://user-images.githubusercontent.com/1188538/136822633-1cd9a3c7-5afb-43b8-a3cd-7a44f8003cb7.png)

With the extra settings the scaling is as desired, but all the embedded images are missing:

![image](https://user-images.githubusercontent.com/1188538/136823835-4f88e1a9-631e-40e3-988f-5ef116ba7381.png)

<details><summary>Here is the complete code</summary>

```kotlin
                val locationNumber = "2-2192362"

                val imageView = root.findViewById<ImageView>(R.id.weatherGraph)

                val resp: HttpResponse = ktorHttpClient.request {
                    method = HttpMethod.Get
                    url.takeFrom("https://www.yr.no/en/content/$locationNumber/meteogram.svg")
                    header("User-Agent", "[email address redacted]")
                }

                val svg = SVG.getFromString(resp.readText())

                val b = Bitmap.createBitmap(imageView.width, imageView.height, Bitmap.Config.ARGB_8888)
                val c = Canvas(b)
                c.drawRGB(255, 255, 255)
                svg.setDocumentViewBox(0f, 0f, svg.documentWidth, svg.documentHeight);
                svg.setDocumentHeight("100%")
                svg.setDocumentWidth("100%")
                svg.renderToCanvas(c)

                imageView.setImageBitmap(b)
```
</details>

**Comment by AndreKR** [comment]
I looked into this a bit and I noticed something...

So, `svg.setDocumentViewBox(0f, 0f, svg.documentWidth, svg.documentHeight);` is enough to trigger the issue.

Those little icons look like this in the source:
![image](https://user-images.githubusercontent.com/1188538/180581188-7f626e10-4feb-406f-a5cd-f122f50bf025.png)

These three function are involved:

https://github.com/BigBadaboom/androidsvg/blob/dbe1a8b44339daea2b8bb9b8abed219db6ed4563/androidsvg/src/main/java/com/caverock/androidsvg/utils/SVGAndroidRenderer.java#L639-L645

https://github.com/BigBadaboom/androidsvg/blob/dbe1a8b44339daea2b8bb9b8abed219db6ed4563/androidsvg/src/main/java/com/caverock/androidsvg/utils/SVGAndroidRenderer.java#L703-L714

https://github.com/BigBadaboom/androidsvg/blob/dbe1a8b44339daea2b8bb9b8abed219db6ed4563/androidsvg/src/main/java/com/caverock/androidsvg/utils/SVGAndroidRenderer.java#L291-L297

When I didn't call `setDocumentViewBox()` then the call to `render()` in line 644 would have the `viewPort` parameter set to `[0.0 0.0 24.0 24.0]`. (This seems to make sense because the icon is to be rendered into a space 24x24 in size?)

But when I called `setDocumentViewBox()` then the call to `render()` would have `viewPort` set to `[0.0 0.0 782.0 391.0]`. (That seems to be the full size of the document.) This is because in the error case, `state.viewBox` in `getCurrentViewPortInUserUnits()`  is not null, unlike in the working case where it is null.

Unfortunately this is where I'm a bit stuck because I don't know which state exactly `state` is supposed to represent and why the document viewbox might "leak" down into the viewport of sub SVG elements.
